sklearn/utils/estimator_checks.py

```diff
@@ -713,6 +713,9 @@ def _set_checking_parameters(estimator):
     if name in CROSS_DECOMPOSITION:
         estimator.set_params(n_components=1)
 
+    if name == "GenericUnivariateSelect":
+        estimator.set_params(score_func=f_regression)
```

Comment:
Does the common test will pass with the default value for `score_func=f_classif`

If so, there are two separate issues here:

1. `score_func=f_regression` should pass common tests
2. Preserve dtypes for `GenericUnivariateSelect`

I think we should focus on 2 in this PR, to reduce the scope.

Reply:
It indeed passes when `score_func=f_classif`. In which case, will it still be fine to merge this PR despite the failing test?

Reply:
We can revert this change, since the default is `f_classif`.